src/index.ts

```diff
@@ -212,21 +180,75 @@ function _decode(input: Buffer): Decoded {
   }
 }
 
-/** Check if a string is prefixed by 0x */
+// Check if a string is prefixed by 0x
 function isHexPrefixed(str: string): boolean {
-  return str.slice(0, 2) === '0x'
+  return str.length > 2 && str[0] === '0' && str[1] === 'x'
```

Comment:
should the length check be `>= 2` so `0x` returns true, to represent an empty array? this is commonly used like this `isHexPrefixed(str) ? str : addHexPrefix(str)` and that might turn `0x` into `0x0x`